vignettes/betareg.Rmd

```diff
@@ -0,0 +1,195 @@
+__rstanarm__ Package"](rstanarm.html). This vignette focuses on Step 1 when the likelihood is
+the product of beta distributions.
+
+# Likelihood
+
+Beta regression uses the beta distribution as the likelihood for the data,
+$$
+f(y_i | \alpha,\beta) = \frac{x_i^{(\alpha-1)}(1-x_i)^{(\beta-1)}}{B(\alpha,\beta)}
+$$
+where $B(\cdot)$ is the beta function. The shape parameters for the distribution are $\alpha$ and $\beta$. The shape parameters enter
+into the model according to the following transformations.
+$$
+\alpha = \mu\cdot\phi \\
+\beta = (1-\mu)\cdot\phi
+$$
+
+Where $g_1(\mu) = \mathbf{X}\boldsymbol{\eta}$ where $g_1(\cdot)$ is some link function, $X$ is a $n\times K$ dimensional matrix of
+predictors, and $\eta$ is a $K$ dimensional vector of parameters associated with each predictor. 
+
+In the simplest case (with only one set of regressors), $\phi$ is a scalar parameter. Alternativley, it
+is possible to model $\phi$ using a second set of regressors $Z$, in which case 
+$g_2(\phi) = \mathbf{Z}\boldsymbol{\gamma}$, where $g_2(\cdot)$ is some link function and $\boldsymbol{\gamma}$ is a $L$
+dimensional vector of parameters associated with the $n\times L$ dimensional predictor matrix $Z$. 
+
+# Priors
+
+```{r, child="children/stan_glm_priors.txt"}
+```
+
+# Posterior
+
+When using only a *single set of regressors*, the posterior distribution of $\eta$ and $\phi$ is
+proportional to the product of the likelihood contributions, priors on the $K$ $\eta$ parameters, and $\phi$,
+$$
+f(\boldsymbol{\eta},\phi|\mathbf{y},\mathbf{X}) \propto 
+\prod_{i=1}^n f(y_i | \alpha,\beta) \times 
+\prod_{i=1}^K f(\boldsymbol{\eta}) \times
+f(\phi)
+$$
+
+When using *two sets of regressors*, the posterior distribution of $\eta$ and $\gamma$ is
+proportional to the product of the likelihood contribution, the $K$ priors on the $\eta$ parameters, and the
+$L$ priors on the $\gamma$ parameters,
+
+$$
+f(\boldsymbol{\eta},\boldsymbol{\gamma}|\mathbf{y},\mathbf{X}) \propto 
+\prod_{i=1}^n f(y_i | \alpha,\beta) \times 
+\prod_{k=1}^K f(\boldsymbol{\eta}) \times
+\prod_{l=1}^L f(\boldsymbol{\gamma})
+$$
+
+# An Example Using Simulated Data
+
+In this example the outcome variable $\mathbf{y}$ is simulated in a way that warrants the use of beta
+regression. It is worth mentioning that the data generation process is quite convoluted,
+which is apparent in the identification of the likelihood above. 
+
+The data simulated below uses the logistic link function on the first set of regressors
+and the log link function on the second set of regressors. 
+
+```{r simulated-data, fig.height=5}
+dat <- list()
+dat$eta <- c(1, -0.2)
+dat$gamma <- c(1.8, 0.4)
+dat$N <- 500
+dat$x <- rnorm(dat$N, 2, 2)
+dat$z <- rnorm(dat$N, 2, 2)
+dat$mu <- binomial(link = logit)$linkinv(dat$eta[1] + dat$eta[2]*dat$x)
+dat$phi <- binomial(link = log)$linkinv(dat$gamma[1] + dat$gamma[2]*dat$z)
+dat$y <- rbeta(dat$N, dat$mu * dat$phi, (1 - dat$mu) * dat$phi)
```

Comment:
Same question as my question about this in the example for `stan_betareg.R`. Why the `dat` if it just means more work and code clutter calling `dat$` every time you want to do anything?